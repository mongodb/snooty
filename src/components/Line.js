import React from 'react';

const Line = () => (
  <div className="line">
    <br />
  </div>
);

export default Line;
