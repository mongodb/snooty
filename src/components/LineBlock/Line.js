import React from 'react';
import PropTypes from 'prop-types';
import ComponentFactory from '../ComponentFactory';

const Line = ({ nodeData: { children }, ...rest }) => {
  if (children.length !== 0) {
    return (
      <div className="line">
        {children.map((child, index) => (
          <ComponentFactory key={index} {...rest} nodeData={child} />
        ))}
      </div>
    );
  }
  return (
    <div className="line">
      <br />
    </div>
  );
};

Line.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
};
export default Line;
