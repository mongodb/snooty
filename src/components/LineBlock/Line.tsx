import React from 'react';
import ComponentFactory from '../ComponentFactory';
import { ParentNode } from '../../types/ast';

export type LineProps = {
  nodeData: ParentNode;
};

const Line = ({ nodeData: { children }, ...rest }: LineProps) => {
  if (children.length !== 0) {
    return (
      <div className="line">
        {children.map((child, index) => (
          <ComponentFactory key={index} {...rest} nodeData={child} />
        ))}
      </div>
    );
  }
  return (
    <div className="line">
      <br />
    </div>
  );
};

export default Line;
