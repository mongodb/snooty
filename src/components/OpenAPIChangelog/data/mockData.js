export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
