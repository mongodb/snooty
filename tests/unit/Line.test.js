import React from 'react';
import { render } from '@testing-library/react';
import Line from '../../src/components/LineBlock/Line';

// data for this component
import mockData from './data/Line.test.json';
import mockData2 from './data/Line-empty.test.json';

it('renders correctly', () => {
  const tree = render(<Line nodeData={mockData} />);
  expect(tree.asFragment()).toMatchSnapshot();
});

it('renders an empty Line node correctly', () => {
  const tree = render(<Line nodeData={mockData2} />);
  expect(tree.asFragment()).toMatchSnapshot();
});
