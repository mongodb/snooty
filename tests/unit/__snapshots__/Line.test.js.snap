// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an empty Line node correctly 1`] = `
<DocumentFragment>
  <div
    class="line"
  >
    <br />
  </div>
</DocumentFragment>
`;

exports[`renders correctly 1`] = `
<DocumentFragment>
  .emotion-2 {
  background-color: #F9FBFA;
  border: 1px solid #B8C4C2;
  border-radius: 3px;
  font-family: 'Source Code Pro',Menlo,monospace;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  white-space: nowrap;
}

[data-leafygreen-ui="anchor-inline-code"]:hover > .emotion-2 {
  box-shadow: 0 0 0 3px #E7EEEC;
}

[data-leafygreen-ui="anchor-inline-code"]:focus > .emotion-2 {
  box-shadow: 0 0 0 3px #C5E4F2;
}

.emotion-0 {
  font-size: unset;
}

<div
    class="line"
  >
    <code
      class="emotion-0 emotion-1 emotion-2"
    >
      N. Virginia (us-east-1)
    </code>
  </div>
</DocumentFragment>
`;
